internal/provider/resource_group_members.go

```diff
@@ -192,9 +191,9 @@ func resourceGroupMembersRead(ctx context.Context, d *schema.ResourceData, meta
 	}
 
 	groupId := d.Get("group_id").(string)
-
+	includeDerivedMembership := d.Get("include_derived_membership").(bool)
```

Comment:
```suggestion
	// include_derived_membership is a read-only option available in the datasource, but not in the resource
	// because the datasource and resource share the same Read function, we can add it in here, but need to
	// make sure it is always false for the resource
	includeDerivedMembership := false
	if includeDM, ok := d.GetOk("include_derived_membership"); ok {
		includeDerivedMembership = includeDM.(bool)
	}
```
I've included a comment here for future reference. Technically, the read functions should be separate functions, I think. I'm comfortable with the data source and resource sharing the Read function so long as there is only one difference between the to.

Reply:
Yeah it's tough when an API makes certain fields only applicable to read calls and not to write. I've seen other providers that share read functions though, so I think we're good haha.

Also, I'm happy to commit your suggestion, but I'm curious, why the need for `GetOk` and variable declaration when we have to cast to a `bool` anyway? Even if the option is empty, it will cast to `false`, which is the default option, right? Or am I missing an edge case?

Reply:
Good question! If it's not defined in the config, it will be `nil` and cannot be cast into a `bool`.
```
panic: interface conversion: interface {} is nil, not bool
```

Reply:
Ohh cool! Great catch, thanks for the explanation :)